### PR TITLE
Add sample for toolchains

### PR DESCRIPTION
**Pull request by bmuskalla**
Uses a multi-module project to showcase setting up
toolchains using a convention plugin as well
as providing a custom toolchain spec for individual
tasks.

### CODE REVIEW

**Comment by bmuskalla** [comment]
@bot-gradle test QF

**Comment by bmuskalla** [comment]
@bot-gradle test QF

**Comment by bmuskalla** [comment]
@bot-gradle test this

**Comment by ljacomet** [comment]
Looks like there is a test error caused by accessing the download URL at configuration time. Working on fixing that.

**Comment by ljacomet** [comment]
@bot-gradle test QF

File: subprojects/docs/src/samples/java/jvm-multi-project-with-toolchains/README.adoc [hunk 1]
```diff
@@ -0,0 +1,29 @@
+NOTE: You can open this sample inside an IDE using the https://www.jetbrains.com/help/idea/gradle.html#gradle_import_project_start[IntelliJ native importer] or https://projects.eclipse.org/projects/tools.buildship[Eclipse Buildship].
+
+This sample shows how to configure a toolchain for a JVM project in Gradle.
+Your project usually targets a specific java version.
+Using toolchains, it is very simple to set the required java version while Gradle handles the setting up the tasks (e.g. compile and test) accordingly.
```
Comment (ljacomet) [inline subprojects/docs/src/samples/java/jvm-multi-project-with-toolchains/README.adoc:5]: > ... while Gradle sets up the tasks ...

Removed not needed words

File: subprojects/docs/src/samples/java/jvm-multi-project-with-toolchains/README.adoc [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+Concretely, we add a _convention plugin_ in `buildSrc` to share the setup between multiple subprojects:
+
+====
+include::sample[dir="groovy",files="buildSrc/src/main/groovy/myproject.java-conventions.gradle[tags=toolchain]"]
+include::sample[dir="kotlin",files="buildSrc/src/main/kotlin/myproject.java-conventions.gradle.kts[tags=toolchain]"]
+====
+
+While most modules are fine with the defaults, the one of our modules overrides the defaults as it has more specific requirements.
```
Comment (ljacomet) [inline subprojects/docs/src/samples/java/jvm-multi-project-with-toolchains/README.adoc:14]: > ... defaults, one of our modules ...

Removed extra "the"

**Review by ljacomet — APPROVED** [review]
Looks good

Will take care of the two minor edits below

**Comment by ljacomet** [comment]
Closing as #14474 includes these changes

**Pull request closed without merging**